spec/codelog/command/new_spec.rb

```diff
@@ -2,53 +2,51 @@
 
 describe Codelog::Command::New do
   describe '#run' do
+    let(:name) { 'change' }
+    let(:options) { {} }
+
+    subject { described_class.new name, options }
+    
     before :each do
       allow(Time).to receive_message_chain(:now, :strftime) { '20180119134323984' }
       allow(subject).to receive(:puts)
       allow(subject).to receive(:system) { true }
-      subject.run(options)
+      subject.run
     end
 
-    context 'with no additional options' do
-      let(:options) { Hash.new }
-
-      it 'prints a message to notify the user about the file creation' do
-        expect(subject).to have_received(:puts)
+    it 'prints a message to notify the user about the file creation' do
+      expect(subject).to have_received(:puts)
         .with('== Creating changelogs/unreleased/20180119134323984_change.yml change file based on example ==')
-      end
+    end
 
-      it 'creates a file for the unreleased partial changes' do
-        expect(subject).to have_received(:system)
+    it 'creates a file for the unreleased partial changes' do
+      expect(subject).to have_received(:system)
         .with('cp changelogs/template.yml changelogs/unreleased/20180119134323984_change.yml')
-      end
     end
 
     context "with 'edit' option" do
-      let(:options) {{ edit: true }}
-
-      it 'prints a message to notify the user about the file creation' do
-        expect(subject).to have_received(:puts)
-        .with('== Creating changelogs/unreleased/20180119134323984_change.yml change file based on example ==')
-      end
+      let(:options) { { edit: true } }
 
-      it 'creates a file for the unreleased partial changes' do
+      it 'opens the default text editor with the created file' do
         expect(subject).to have_received(:system)
-        .with('cp changelogs/template.yml changelogs/unreleased/20180119134323984_change.yml')
+          .with('${VISUAL:-${EDITOR:-nano}} changelogs/unreleased/20180119134323984_change.yml')
       end
+    end
 
-      it 'opens the default text editor with the created file' do
+    context 'with a name in a non snake case format' do
+      let(:name) { 'TestName' }
+
+      it 'converts the string to snake case notation, creating the file using it' do
         expect(subject).to have_received(:system)
-        .with('${VISUAL:-${EDITOR:-nano}} changelogs/unreleased/20180119134323984_change.yml')
-      end
+          .with('cp changelogs/template.yml changelogs/unreleased/20180119134323984_test_name.yml')
+      end 
     end
   end
 
   describe '.run' do
```

Comment:
Testing simple wrapper methods is not worth it, in my opinion. They offer no documental value.

Besides, it is strange (for me) to leave both `#run` and `.run` as public API. If this class is supposed to be a "MethodObject", I think that leaving only the class method as public API would be sufficient.

Reply:
I won't change this in this PR, because we did it all the time in the tests. If you want to change it, please open an issue. As for the API, it's like an alias for the method. I don't see where this could be a problem for us.